api/graphite.go

```diff
@@ -339,53 +388,63 @@ func (s *Server) metricsIndex(ctx *middleware.Context) {
 		response.Write(ctx, response.WrapError(err))
 		return
 	}
-	errors := make([]error, 0)
-	series := make([]idx.Archive, 0)
-	seenDefs := make(map[string]struct{})
-	var mu sync.Mutex
+	reqCtx := ctx.Req.Context()
+	responses := make(chan struct {
```

Comment:
afaik types can have a function scope, so it might be worth it to declare a type for that response struct instead of redefining it 3 times